LLVM 23.1 moved HAVE_BACKTRACE/BACKTRACE_HEADER and HAVE_MALLINFO behind
selects keyed on @llvm//platforms/config constraint settings from the
hermetic-llvm BCR module. TAPA's tapacc build does not depend on that
module; restore the LLVM 22 behavior of defining them unconditionally
(execinfo.h/mallinfo exist on the glibc Linux and macOS hosts tapacc runs
on; the musl static tools only compile, they never execute this code).
diff --git a/utils/bazel/llvm-project-overlay/llvm/config.bzl b/utils/bazel/llvm-project-overlay/llvm/config.bzl
index 08167cb..d592f9e 100644
--- a/utils/bazel/llvm-project-overlay/llvm/config.bzl
+++ b/utils/bazel/llvm-project-overlay/llvm/config.bzl
@@ -28,6 +28,8 @@ def native_arch_defines(arch, triple):
 
 posix_defines = [
     "LLVM_ON_UNIX=1",
+    "HAVE_BACKTRACE=1",
+    "BACKTRACE_HEADER=<execinfo.h>",
     r'LTDL_SHLIB_EXT=\".so\"',
     r'LLVM_PLUGIN_EXT=\".so\"',
     "LLVM_ENABLE_LLVM_EXPORT_ANNOTATIONS=1",
@@ -46,23 +48,10 @@ posix_defines = [
     "HAVE_UNISTD_H=1",
 ]
 
-backtrace_defines = select({
-    "@platforms//os:windows": [],
-    "@llvm//platforms/config:musl": [],
-    "//conditions:default": [
-        "HAVE_BACKTRACE=1",
-        "BACKTRACE_HEADER=<execinfo.h>",
-    ],
-})
-
-mallinfo_defines = select({
-    "@llvm//platforms/config:gnu": ["HAVE_MALLINFO=1"],
-    "//conditions:default": [],
-})
-
 linux_defines = posix_defines + [
     "_GNU_SOURCE",
     "HAVE_GETAUXVAL=1",
+    "HAVE_MALLINFO=1",
     "HAVE_SBRK=1",
     "HAVE_STRUCT_STAT_ST_MTIM_TV_NSEC=1",
 ]
@@ -98,7 +87,7 @@ os_defines = select({
     "@platforms//os:macos": macos_defines,
     "@platforms//os:windows": win32_defines,
     "//conditions:default": linux_defines,
-}) + backtrace_defines + mallinfo_defines
+})
 
 # HAVE_BUILTIN_THREAD_POINTER is true for on Linux (outside of ppc64) for
 # all recent toolchains. Add it here by default on Linux as we can't perform a
